src/RestQuery.js

```diff
@@ -335,7 +335,7 @@ RestQuery.prototype.replaceNotInQuery = function() {
 const transformSelect = (selectObject, key ,objects) => {
   var values = [];
   for (var result of objects) {
-    values.push(result[key]);
+    values.push(key.split('.').reduce((o,i)=>o[i], result));
```

Comment:
If we change this for `transformSelect` we should also change the similar code in `transformDontSelect`. Additionally another test should then be added to account for `doesNotMatchKeyInQuery` having this modified behavior as well.

Reply:
Ah, yes, totally missed that. I'll give it a go soon (tm)